File: Backend/Travle.Model/Responses/DashboardResponse.cs
namespace Travle.Model.Responses
{
    /// <summary>
    /// The admin dashboard payload (course §2.4): headline metric tiles, the bookings-per-month chart
    /// series, and the recent-activity feed. All figures are computed server-side with DB-level
    /// aggregates over the whole platform. Money is in <see cref="Currency"/> minor-of-none decimals
    /// (KM), matching the payments screen.
    /// </summary>
    public class DashboardResponse
    {
        /// <summary>Total registered user accounts.</summary>
        public int TotalUsers { get; set; }

        /// <summary>
        /// Active tours: <c>IsActive</c>, owned by a non-suspended organizer, with at least one future
        /// non-cancelled schedule.
        /// </summary>
        public int ActiveTours { get; set; }

        /// <summary>Role applications still awaiting an admin decision.</summary>
        public int PendingRoleApplications { get; set; }

        /// <summary>Destinations awaiting moderation.</summary>
        public int PendingDestinations { get; set; }

        /// <summary>Net revenue captured in the current calendar month (gross captured − refunds issued).</summary>
        public decimal MonthlyNetRevenue { get; set; }

        public string Currency { get; set; } = "bam";

        /// <summary>Bookings per month for the trailing 12 months (oldest first, no gaps).</summary>
        public List<MonthlyBookingPoint> BookingsPerMonth { get; set; } = new();

        /// <summary>Most-recent platform events, newest first.</summary>
        public List<DashboardActivityItem> RecentActivity { get; set; } = new();
    }
}

File: UI/travle_core/lib/src/models/dashboard_response.dart
import 'package:json_annotation/json_annotation.dart';

import 'dashboard_activity_item.dart';
import 'monthly_booking_point.dart';

part 'dashboard_response.g.dart';

/// The admin dashboard payload (mirrors the backend `DashboardResponse`): headline
/// metric tiles, the bookings-per-month chart series and the recent-activity feed.
/// Money is in [currency] (displayed as "KM").
@JsonSerializable()
class DashboardResponse {
  DashboardResponse({
    required this.totalUsers,
    required this.newUsersThisMonth,
    required this.totalBookings,
    required this.activeTours,
    required this.pendingRoleApplications,
    required this.pendingDestinations,
    required this.monthlyNetRevenue,
    required this.currency,
    required this.bookingsPerMonth,
    required this.recentActivity,
  });

  final int totalUsers;
  final int newUsersThisMonth;
  final int totalBookings;
  final int activeTours;
  final int pendingRoleApplications;
  final int pendingDestinations;
  final double monthlyNetRevenue;
  final String currency;
  final List<MonthlyBookingPoint> bookingsPerMonth;
  final List<DashboardActivityItem> recentActivity;

  factory DashboardResponse.fromJson(Map<String, dynamic> json) =>
      _$DashboardResponseFromJson(json);

  Map<String, dynamic> toJson() => _$DashboardResponseToJson(this);
}
